include/circt/Dialect/Emit/EmitOps.td

```diff
@@ -0,0 +1,99 @@
+//===- EmitOps.td - `emit` dialect ops ---------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the MLIR ops for `emit`.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_EMIT_EMITOPS_TD
+#define CIRCT_DIALECT_EMIT_EMITOPS_TD
+
+include "circt/Dialect/HW/HWTypes.td"
+include "circt/Dialect/Emit/EmitDialect.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+class EmitOp<string mnemonic, list<Trait> traits = []> :
+    Op<EmitDialect, mnemonic, traits>;
+
+def FileOp : EmitOp<"file", [
```

Comment:
We don't want to capture SSA values outside, correct? If so could you add `IsolatedFromAbove`?